flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/streamrecord/StreamElementSerializerTest.java

```diff
@@ -89,11 +155,59 @@ public void testSerialization() throws Exception {
         Watermark negativeWatermark = new Watermark(-4647654567676555876L);
         assertEquals(negativeWatermark, serializeAndDeserialize(negativeWatermark, serializer));
 
+        final WatermarkStatus idle = new WatermarkStatus(IDLE_STATUS);
+        assertEquals(idle, serializeAndDeserialize(idle, serializer));
+
+        final WatermarkStatus active = new WatermarkStatus(ACTIVE_STATUS);
+        assertEquals(active, serializeAndDeserialize(active, serializer));
+
         LatencyMarker latencyMarker =
                 new LatencyMarker(System.currentTimeMillis(), new OperatorID(-1, -1), 1);
         assertEquals(latencyMarker, serializeAndDeserialize(latencyMarker, serializer));
     }
 
+    @Test
+    public void testSerializationWithReuse() throws Exception {
+        final StreamElementSerializer<String> serializer =
+                new StreamElementSerializer<>(StringSerializer.INSTANCE);
+        final StreamRecord<String> reuse = new StreamRecord<>("");
+
+        StreamRecord<String> withoutTimestamp = new StreamRecord<>("test 1 2 分享基督耶穌的愛給們，開拓雙贏!");
+        assertEquals(
+                withoutTimestamp,
+                serializeAndDeserializeWithReuse(withoutTimestamp, serializer, reuse));
+
+        StreamRecord<String> withTimestamp = new StreamRecord<>("one more test 拓 們 分", 77L);
+        assertEquals(
+                withTimestamp, serializeAndDeserializeWithReuse(withTimestamp, serializer, reuse));
+
+        StreamRecord<String> negativeTimestamp = new StreamRecord<>("他", Long.MIN_VALUE);
+        assertEquals(
+                negativeTimestamp,
+                serializeAndDeserializeWithReuse(negativeTimestamp, serializer, reuse));
```

Comment:
I'd also like to see an assert that `reuse` is actually returned.

You could switch to assertj-style assertions to make it more beautiful:
```
assertThat(serializeAndDeserializeWithReuse(negativeTimestamp, serializer, reuse))
  .isEqualTo(negativeTimestamp)
  .isSameAs(reuse)
```